In [10]:
today = 20190328
dataframe = pd.read_csv('./Crawled Data/{}/{}_KTR_KTS_dataframe'.format(today, today))
dataframe = dataframe.sort_values(by='Total_KTR', ascending=False).iloc[0:20]
dataframe = dataframe.drop_duplicates(subset = "Keyword")
dataframe

Topic Keyword  Total_KTR  Topic_KTR  KTS  \
11  politics     박영선        487        268   22   
20  economic     조양호        374        266   46   
18  politics     황교안        354        195   34   
15  politics     김학의        340        188   21   
24  economic     경영권        187        133   63   
1    society    대한항공        184        133   49   
23  economic      연임        183        130   42   
14  politics      의원        147         81   44   
72   digital      KT        143        166   54   
3    society      사장        133         96   45   
29  economic    국민연금        132         93   30   
10  politics      재산        111         61   52   
9    society      국회        104         75   17   
27  economic      부결        104         74   63   
2    society     후보자         92         67   35   

                       Related_Keywords Total_KTR_change KTS_change  
11            ['김의겸', '박영선 황교안', '황교안']             487%        22%  
20    ['조양호 가족', '스튜어드 십 코드', '조양호 나이']             374%        46%  
18     ['박영선 황교안', '박영선', '황교안 법무부 장관']             354%        34%  
15   ['김학 의 cd', '김학 의 동영상 내용', '뱀부 즐']             340%        21%  
24                                 [없음]             187%        63%  
1     ['셀트리온 주가', '일 신석재 주가', '차 바이오텍']             184%        49%  
23                           ['조양호 연임']             183%        42%  
14      ['김병관 의원', '김세연 의원', '이용 주 의원']           147% 🔺      44% 🔻  
72      ['kt 인턴', 'sk 하이닉스 주식', '티 월드']           143% 🔻      54% 🔺  
3          ['모래 사장', '바지 사장', '이부진 사장']             133%        45%  
29  ['국세청', '국민 연금 담보 대출', '스튜어드 십 코드']           132% 🔻      30% 🔺  
10        ['김병관 재산', '박원순 재산', '조국 재산']             111%        52%  
9         ['국회 속기록', '국회 의장', '국회 본회의']             104%        17%  
27                                 [없음]             104%        63%  
2                               ['조동호']            92% 🔻      35% 🔺

In [286]:
main_text = "[📰Daily Bigdata , 3/26]\n\n[🌏오늘의 키워드🌏]\n\n\
    1. {}\n\
    2. {}\n\
    3. {}\n\
    4. {}\n\
    5. {}\n\
    6. {}\n\
    7. {}\n\
    8. {}\n\
    9. {}\n\
    10. {}\n\n\
[💵오늘의 증시💵]\n\
코스피 2,230.50(+1.84)\n\
코스닥 743.38(-3.95)\n\
환   율 1,125.0(-1.0)\n\
\n\
📌영상으로 보고 싶으시다면?\n\
     데일리 빅데이터 유튜브!!📌\n\
    (링크)\n\n\
🚨 데일리 빅데이터를 이용한 투자 피해에는 책임을 지지 않으며 🚨\n\
        분석 자료의 해석은 개인마다 다를 수 있습니다.".format(total_top10[0],total_top10[1],total_top10[2],total_top10[3],\
                  total_top10[4],total_top10[5],total_top10[6],total_top10[7],\
                  total_top10[8],total_top10[9])


In [287]:
print(main_text)

[📰Daily Bigdata , 3/26]

[🌏오늘의 키워드🌏]

    1. 의혹
    2. 한국
    3. 손흥민
    4. 후보자
    5. 콜롬비아
    6. 장관
    7. 5명
    8. 대한항공
    9. 국민연금
    10. 의사

[💵오늘의 증시💵]
코스피 2,230.50(+1.84)
코스닥 743.38(-3.95)
환   율 1,125.0(-1.0)

📌영상으로 보고 싶으시다면?
     데일리 빅데이터 유튜브!!📌
    (링크)

🚨 데일리 빅데이터를 이용한 투자 피해에는 책임을 지지 않으며 🚨
        분석 자료의 해석은 개인마다 다를 수 있습니다.


In [221]:
top10_topic = "[📰Daily Bigdata , 3/26]\n\n[🌏오늘의 키워드🌏]\n\n\
    1. {}\n\
    2. {}\n\
    3. {}\n\
    4. {}\n\
    5. {}\n\
    6. {}\n\
    7. {}\n\
    8. {}\n\
    9. {}\n\
    10. {}\n\n\
[💵오늘의 증시💵]\n\
코스피 2,230.50(+1.84)\n\
코스닥 743.38(-3.95)\n\
환   율 1,125.0(-1.0)\n\
\n\
📌영상으로 보고 싶으시다면?\n\
     데일리 빅데이터 유튜브!!📌\n\
    (링크)\n\
\n\
[🔎키워드 분석🔍]\n\
1. {}\n\
키워드 관심도 : {}% 🔺️\n\
키워드 감정도 : {}% 🔺️\n\
관련뉴스 : (링크) \n\
연관 검색 : {}\n\
\n\
2. {}\n\
키워드 관심도 : {}% 🔺️\n\
키워드 감정도 : {}% 🔺️\n\
관련뉴스 : 링크\n\
연관검색 : {}\n\
\n\
3. {}\n\
키워드 관심도 : {}% 🔺️\n\
키워드 감정도 : {}% 🔺️\n\
관련뉴스 : (링크) \n\
연관 검색 : {}\n\
\n\
4. {}\n\
키워드 관심도 : {}% 🔺️\n\
키워드 감정도 : {}% 🔺️\n\
관련뉴스 : 링크\n\
연관검색 : {}\n\
\n\
5. {}\n\
키워드 관심도 : {}% 🔺️\n\
키워드 감정도 : {}% 🔺️\n\
관련뉴스 : (링크) \n\
연관 검색 : {}\n\
\n\
6. {}\n\
키워드 관심도 : {}% 🔺️\n\
키워드 감정도 : {}% 🔺️\n\
관련뉴스 : 링크\n\
연관검색 : {}\n\
\n\
7. {}\n\
키워드 관심도 : {}% 🔺️\n\
키워드 감정도 : {}% 🔺️\n\
관련뉴스 : (링크) \n\
연관 검색 : {}\n\
\n\
8. {}\n\
키워드 관심도 : {}% 🔺️\n\
키워드 감정도 : {}% 🔺️\n\
관련뉴스 : 링크\n\
연관검색 : {}\n\
\n\
9. {}\n\
키워드 관심도 : {}% 🔺️\n\
키워드 감정도 : {}% 🔺️\n\
관련뉴스 : (링크) \n\
연관 검색 : {}\n\
\n\
10. {}\n\
키워드 관심도 : {}% 🔺️\n\
키워드 감정도 : {}% 🔺️\n\
관련뉴스 : 링크\n\
연관검색 : {}\n\
\n\
[💵오늘의 증시💵]\n\
코스피 2,230.50(+1.84)\n\
코스닥 743.38(-3.95)\n\
환   율 1,125.0(-1.0)\n\
\n\
📌영상으로 보고 싶으시다면?\n\
     데일리 빅데이터 유튜브!!📌\n\
    (링크)".format(total_top10[0],total_top10[1],total_top10[2],total_top10[3],\
                  total_top10[4],total_top10[5],total_top10[6],total_top10[7],\
                  total_top10[8],total_top10[9],\
                  total_top10[0],total_top10_KTR[0],total_top10_KTS[0],total_top10_RK[0],\
                  total_top10[1],total_top10_KTR[1],total_top10_KTS[1],total_top10_RK[1],\
                  total_top10[2],total_top10_KTR[2],total_top10_KTS[2],total_top10_RK[2],\
                  total_top10[3],total_top10_KTR[3],total_top10_KTS[3],total_top10_RK[3],\
                  total_top10[4],total_top10_KTR[4],total_top10_KTS[4],total_top10_RK[4],\
                  total_top10[5],total_top10_KTR[5],total_top10_KTS[5],total_top10_RK[5],\
                  total_top10[6],total_top10_KTR[6],total_top10_KTS[6],total_top10_RK[6],\
                  total_top10[7],total_top10_KTR[7],total_top10_KTS[7],total_top10_RK[7],\
                  total_top10[8],total_top10_KTR[8],total_top10_KTS[8],total_top10_RK[8],\
                  total_top10[9],total_top10_KTR[9],total_top10_KTS[9],total_top10_RK[9])

file = open("./Crawled Data/{}/{}_top10_text".format(today,today), "w")
file.write(top10_topic)
file.close()

In [215]:
print(top10_topic)

[📰Daily Bigdata , 3/26]

[🌏오늘의 키워드🌏]

    1. 의혹
    2. 한국
    3. 손흥민
    4. 후보자
    5. 후보자
    6. 콜롬비아
    7. 장관
    8. 장관
    9. 5명
    10. 대한항공

[💵오늘의 증시💵]
코스피 2,230.50(+1.84)
코스닥 743.38(-3.95)
환   율 1,125.0(-1.0)

📌영상으로 보고 싶으시다면?
     데일리 빅데이터 유튜브!!📌
    (링크)

[🔎키워드 분석🔍]
1. 의혹
키워드 관심도 : 309% 🔺️
키워드 감정도 : 19% 🔺️
관련뉴스 : (링크) 
연관 검색 : #없음

2. 한국
키워드 관심도 : 269% 🔺️
키워드 감정도 : 38% 🔺️
관련뉴스 : 링크
연관검색 : #한국 콜롬비아 축구 다시 보기 #한국 콜롬비아 다시 보기 #한국 콜롬비아 토렌트

3. 손흥민
키워드 관심도 : 252% 🔺️
키워드 감정도 : 68% 🔺️
관련뉴스 : (링크) 
연관 검색 : #손흥 민 바니 바니 #콜롬비아 전 #황인범

4. 후보자
키워드 관심도 : 219% 🔺️
키워드 감정도 : 26% 🔺️
관련뉴스 : 링크
연관검색 : #없음

5. 후보자
키워드 관심도 : 216% 🔺️
키워드 감정도 : 26% 🔺️
관련뉴스 : (링크) 
연관 검색 : #없음

6. 콜롬비아
키워드 관심도 : 211% 🔺️
키워드 감정도 : 55% 🔺️
관련뉴스 : 링크
연관검색 : #콜롬비아 오프사이드 #오프사이드 #손흥 민 골

7. 장관
키워드 관심도 : 189% 🔺️
키워드 감정도 : 40% 🔺️
관련뉴스 : (링크) 
연관 검색 : #중소 벤처 기업 부 장관 #이매리 실명 #이매리 명단

8. 장관
키워드 관심도 : 188% 🔺️
키워드 감정도 : 43% 🔺️
관련뉴스 : 링크
연관검색 : #중소 벤처 기업 부 장관 #이매리 실명 #이매리 명단

9. 5명
키워드 관심도 : 176% 🔺️
키워드 감정도 : 48% 🔺️
관련뉴스 : (링크) 
연관 검색 

In [202]:
for i in range(10):
    if total_top10_RK[i] == '[없음]':
        total_top10_RK[i] = '#없음'
    else:
        temp = []
        total_top10_RK[i] = total_top10_RK[i][1:-1]
        total_top10_RK[i] = total_top10_RK[i].split(', ')
        for j in range(len(total_top10_RK[i])):
            temp.append('#'+total_top10_RK[i][j][1:-1])
        temp = (' ').join(temp)
        total_top10_RK[i] = temp

In [98]:
import pandas as pd
from datetime import timedelta,date 
import os

today = 20190328
yesterday = 20190327
today_KTR_KTS = pd.read_csv('./Crawled Data/{}/{}_KTR_KTS_dataframe'.format(today, today))
yesterday_KTR_KTS = pd.read_csv('./Crawled Data/{}/{}_KTR_KTS_dataframe'.format(yesterday, yesterday))

today_KTR_KTS = today_KTR_KTS.drop(columns = ['Total_KTR_change', 'KTS_change'])

In [99]:
Total_KTR_list = today_KTR_KTS['Total_KTR'].values.tolist()
Total_KTS_list = today_KTR_KTS['KTS'].values.tolist()

In [100]:
compare_Total_KTR_list = Total_KTR_list
compare_KTS_list = Total_KTS_list

In [101]:
for i in range(8):

    for j in range(10):

        keyword = today_KTR_KTS.iloc[10*i+j]['Keyword']

        for k in range(10):

            if keyword == yesterday_KTR_KTS.iloc[10*i+k]['Keyword']:

                Total_KTR_change = int(today_KTR_KTS.iloc[10*i+j]['Total_KTR'] - yesterday_KTR_KTS.iloc[10*i+k]['Total_KTR'])
                print(type(Total_KTR_change))
                KTR_change = str(abs(Total_KTR_change))
                
                if(Total_KTR_change < 0):
                    compare_Total_KTR_list[10*i+j] = (str(today_KTR_KTS.iloc[10*i+j]['Total_KTR'])+'% ('+KTR_change+'🔻)')
                else:
                    compare_Total_KTR_list[10*i+j] = (str(today_KTR_KTS.iloc[10*i+j]['Total_KTR'])+'% ('+KTR_change+'🔺)')

                Total_KTS_change = int(today_KTR_KTS.iloc[10*i+j]['KTS'] - yesterday_KTR_KTS.iloc[10*i+k]['KTS'])
                KTS_change = str(abs(Total_KTS_change))
                
                if(Total_KTS_change < 0):
                    compare_KTS_list[10*i+j] = (str(today_KTR_KTS.iloc[10*i+j]['KTS'])+'% ('+KTS_change+'🔻)')
                else:
                    compare_KTS_list[10*i+j] = (str(today_KTR_KTS.iloc[10*i+j]['KTS'])+'% ('+KTS_change+'🔻)')
# %를 붙여줍니다
for i in range(80):
    if ('%' not in str(compare_Total_KTR_list[i])):
        compare_Total_KTR_list[i] = str(compare_Total_KTR_list[i]) + '%'
    if ('%' not in str(compare_KTS_list[i])):
        compare_KTS_list[i] = str(compare_KTS_list[i]) + '%'


today_KTR_KTS['Total_KTR_change'] = compare_Total_KTR_list
today_KTR_KTS['KTS_change'] = compare_KTS_list

<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>
<class 'int'>


In [40]:
today_KTR_KTS

Topic Keyword  Total_KTR  Topic_KTR  KTS  \
0     society    미세먼지         81         57   34   
1     society    대한항공        184        133   49   
2     society     후보자         92         67   35   
3     society      사장        133         96   45   
4     society     조양호        373        270   48   
5     society      용인         70         51   49   
6     society      의원        144        104   49   
7     society      KT        137         99   52   
8     society    민주노총         64         46   47   
9     society      국회        104         75   17   
10   politics      재산        111         61   52   
11   politics     박영선        487        268   22   
12   politics      장관         77         43   37   
13   politics     후보자         91         50   36   
14   politics      의원        147         81   44   
15   politics     김학의        340        188   21   
16   politics    국회의원         72         39   38   
17   politics      공개         75         41   80   
18   politics     황교안        354        195   34   
19   politics      평균         59         33   65   
20   economic     조양호        374        266   46   
21   economic    대한항공        183        130   46   
22   economic    대표이사         69         49   57   
23   economic      연임        183        130   42   
24   economic     경영권        187        133   63   
25   economic     박창진         73         52   83   
26   economic      변경         72         51   65   
27   economic      부결        104         74   63   
28   economic     지부장         30         21   54   
29   economic    국민연금        132         93   30   
..        ...     ...        ...        ...  ...   
50  entertain     한혜진         75        112   77   
51  entertain    골목식당         81        120   57   
52  entertain     류준열         66         98   98   
53  entertain     펜타곤         81        120   86   
54  entertain    프리즈너         54         79  102   
55  entertain      뱅커         52         78  106   
56  entertain      컴백         56         85   99   
57  entertain      닥터         74        110  107   
58  entertain     그레이         71        105   91   
59  entertain   라디오스타         64         94   77   
60     sports      감독         81        119   37   
61     sports    흥국생명         56         84   84   
62     sports     이재영         73        107   90   
63     sports     끝내기         76        112   99   
64     sports      두산         79        117   79   
65     sports     박미희         42         61   82   
66     sports      키움         66         98   56   
67     sports     정수빈         66         98   99   
68     sports      우승         65         96   97   
69     sports      안타         73        108  103   
70    digital      5G         88        103   92   
71    digital      출시         64         75   79   
72    digital      KT        143        166   54   
73    digital     서비스         56         66   75   
74    digital     조동호         80         94   25   
75    digital    대한항공        180        209   48   
76    digital     스마트         57         67   80   
77    digital    대표이사         67         78   52   
78    digital      신임         61         71   64   
79    digital      개발         61         71   64   

                                     Related_Keywords Total_KTR_change  \
0         ['스모그 프리 타워', '미세 먼지 삼겹살', '미세 먼지 원인 과 대책']        81% (16🔻)   
1                   ['셀트리온 주가', '일 신석재 주가', '차 바이오텍']             184%   
2                                             ['조동호']       92% (124🔻)   
3                        ['모래 사장', '바지 사장', '이부진 사장']             133%   
4                         ['조양호 나이', '조양호 재산', '박영선']             373%   
5               ['용인 롯데 몰', '용인 롯데 몰 화재', '용인 수지 화재']              70%   
6                     ['김병관 의원', '김세연 의원', '이용 주 의원']             144%   
7                   ['kt 인턴', 'sk 하이닉스 주식', 'kt 컬러링']             137%   
8                                                [없음]              64%   
9                       ['국회 속기록', '국회 의장',

In [58]:
RK = today_KTR_KTS['Related_Keywords'].values.tolist()

In [59]:
for i in range(80):
    if RK[i] == '[없음]':
        RK[i] = '#없음'
    else:
        temp = []
        RK[i] = RK[i][1:-1]
        RK[i] = RK[i].split(', ')
        for j in range(len(RK[i])):
            RK[i][j] = RK[i][j][1:-1]
            if (" " in RK[i][j]):
                RK[i][j] = RK[i][j].replace(" ", "")
            temp.append('#'+RK[i][j])
            
        temp = (' ').join(temp)
        RK[i] = temp

In [65]:
# 연관검색어 검색
keywords = pd.read_csv('./Crawled Data/{}/{}_Top10_keyword'.format(today, today))
keyword_list = keywords['Keyword'].values.tolist()
top5_topic = []

# pytrend는 keyword를 str이 아니라 list로 받으므로
# top5 topic을 모두 2차원 list로 설정

for i in range(8):
    for j in range(10):
        top5_topic.append([keyword_list[10*i+j]])

# top5 topic의 연관검색어 추출
from pytrends.request import TrendReq

top3_related_keyword = []
top3_related_value = []

pytrends = TrendReq(hl='ko')

for i in range(len(top5_topic)):

    # 시간설정은 지난 하루동안, 지역은 한국설정
    # 'rising'부분이 급상승 키워드 부분입니다(가중치도 함께 출력)

    try:
        pytrends.build_payload(top5_topic[i], geo = 'KR', timeframe='now 1-d')
        queries = pytrends.related_queries()
        dataframe = queries[top5_topic[i][0]]['rising']

        top3_related_keyword.append(dataframe['query'].values.tolist()[0:3])
        top3_related_value.append(dataframe['value'].values.tolist()[0:3])

    except:
        top3_related_keyword.append('[없음]')
        top3_related_value.append('[없음]')
        
for i in range(80):
    
    if top3_related_keyword[i] == '[없음]':
        top3_related_keyword[i] = '#없음'
    else:
        for j in range(len(top3_related_keyword[i])):
            top3_related_keyword[i][j] = '#'+top3_related_keyword[i][j]
            if (" ") in top3_related_keyword[i][j]:
                top3_related_keyword[i][j] = top3_related_keyword[i][j].replace(" ", "")

In [79]:
today = 20190328
dataframe = pd.read_csv('./Crawled Data/{}/{}_KTR_KTS_dataframe'.format(today, today))
dataframe = dataframe.sort_values(by='Total_KTR', ascending=False).iloc[0:20]
dataframe = dataframe.drop_duplicates(subset = "Keyword")
keywords = dataframe.iloc[0:10]['Keyword'].values.tolist()
KTR = dataframe.iloc[0:10]['Total_KTR_change'].values.tolist()
KTS = dataframe.iloc[0:10]['KTS_change'].values.tolist()
related_keywords = dataframe['Related_Keywords'].iloc[0:10].values.tolist()

In [84]:
top1_5_text = "🔎 종합 키워드 분석 TOP1~5\n\
\n\
1. {}\n\
관심도 : {}\n\
감정도 : {}\n\
{}\n\
\n\
2. {}\n\
관심도 : {}\n\
감정도 : {}\n\
{}\n\
\n\
3. {}\n\
관심도 : {}\n\
감정도 : {}\n\
{}\n\
\n\
4. {}\n\
관심도 : {}\n\
감정도 : {}\n\
{}\n\
\n\
5. {}\n\
관심도 : {}\n\
감정도 : {}\n\
{}\n\
\n\
📌영상으로 보고 싶으시다면?\n\
    데일리 빅데이터 유튜브!!📌\n\
    (링크)".format(keywords[0], KTR[0], KTS[0], related_keywords[0],\
                  keywords[1], KTR[1], KTS[1], related_keywords[1],\
                  keywords[2], KTR[2], KTS[2], related_keywords[2],\
                  keywords[3], KTR[3], KTS[3], related_keywords[3],\
                  keywords[4], KTR[4], KTS[4], related_keywords[4])


🔎 종합 키워드 분석 TOP1~5

1. 박영선
관심도 : 487%
감정도 : 22%
['김의겸', '박영선 황교안', '황교안']

2. 조양호
관심도 : 374%
감정도 : 46%
['조양호 가족', '스튜어드 십 코드', '조양호 나이']

3. 황교안
관심도 : 354%
감정도 : 34%
['박영선 황교안', '박영선', '황교안 법무부 장관']

4. 김학의
관심도 : 340%
감정도 : 21%
['김학 의 cd', '김학 의 동영상 내용', '뱀부 즐']

5. 경영권
관심도 : 187%
감정도 : 63%
[없음]

📌영상으로 보고 싶으시다면?
    데일리 빅데이터 유튜브!!📌
    (링크)


In [85]:
top6_10_text = "🔎 종합 키워드 분석 TOP1~5\n\
\n\
6. {}\n\
관심도 : {}\n\
감정도 : {}\n\
{}\n\
\n\
7. {}\n\
관심도 : {}\n\
감정도 : {}\n\
{}\n\
\n\
8. {}\n\
관심도 : {}\n\
감정도 : {}\n\
{}\n\
\n\
9. {}\n\
관심도 : {}\n\
감정도 : {}\n\
{}\n\
\n\
10. {}\n\
관심도 : {}\n\
감정도 : {}\n\
{}\n\
\n\
📌영상으로 보고 싶으시다면?\n\
    데일리 빅데이터 유튜브!!📌\n\
    (링크)".format(keywords[5], KTR[5], KTS[5], related_keywords[5],\
                  keywords[6], KTR[6], KTS[6], related_keywords[6],\
                  keywords[7], KTR[7], KTS[7], related_keywords[7],\
                  keywords[8], KTR[8], KTS[8], related_keywords[8],\
                  keywords[9], KTR[9], KTS[9], related_keywords[9])
print(top6_10_text)

🔎 종합 키워드 분석 TOP1~5

6. 대한항공
관심도 : 184%
감정도 : 49%
['셀트리온 주가', '일 신석재 주가', '차 바이오텍']

7. 연임
관심도 : 183%
감정도 : 42%
['조양호 연임']

8. 의원
관심도 : 147% 🔺
감정도 : 44% 🔻
['김병관 의원', '김세연 의원', '이용 주 의원']

9. KT
관심도 : 143% 🔻
감정도 : 54% 🔺
['kt 인턴', 'sk 하이닉스 주식', '티 월드']

10. 사장
관심도 : 133%
감정도 : 45%
['모래 사장', '바지 사장', '이부진 사장']

📌영상으로 보고 싶으시다면?
    데일리 빅데이터 유튜브!!📌
    (링크)


⚖ 정치 키워드 

    1. 박영선
    2. 황교안
    3. 김학의
    4. 의원
    5. 재산
    6. 후보자
    7. 장관
    8. 공개
    9. 국회의원
    10. 평균


[🔎키워드 분석🔍]

1. 박영선
관심도 : 487%
감정도 : 22%
#김의겸 #박영선황교안 #황교안

2. 황교안
관심도 : 354%
감정도 : 34%
#박영선 황교안 #박영선 #황교안 법무부 장관

3. 김학의
관심도 : 340%
감정도 : 21%
#김학의cd #김학의동영상내용 #뱀부즐

4. 의원
관심도 : 147% 🔺
감정도 : 44% 🔻
#김병관의원 #김세연의원 #이용주의원

5. 재산
관심도 : 111%
감정도 : 52%
#김병관재산 #박원순재산 #조국

In [8]:
today_KTR_KTS = pd.read_csv('./Crawled Data/{}/{}_KTR_KTS_dataframe'.format(today, today))

topics = ['society', 'politics', 'economic', 'foreign', 'culture',
            'entertain', 'sports', 'digital']
topics_emoji = ['🌉','⚖','💲','🌏','🎼','🎤','⚽','💻']

topics_kr = ['사회', '정치', '경제', '국제', '문화', '연예', '스포츠', 'IT']

kr_index = 0

for topic in topics:

    dataframe = today_KTR_KTS[today_KTR_KTS['Topic'] == topic].sort_values(by='Total_KTR', ascending=False)
    keywords = dataframe.iloc[0:10]['Keyword'].values.tolist()
    KTR = dataframe.iloc[0:10]['Total_KTR_change'].values.tolist()
    KTS = dataframe.iloc[0:10]['KTS_change'].values.tolist()
    related_keywords = dataframe['Related_Keywords'].iloc[0:10].values.tolist()

                
        text = "{} {} 키워드\n\n\
    1. {}\n\
    2. {}\n\
    3. {}\n\
    4. {}\n\
    5. {}\n\
    6. {}\n\
    7. {}\n\
    8. {}\n\
    9. {}\n\
    10. {}\n\
\n\n\
[🔎키워드 분석🔍]\n\n\
1. {}\n\
관심도 : {}\n\
감정도 : {}\n\
{}\n\
\n\
2. {}\n\
관심도 : {}\n\
감정도 : {}\n\
{}\n\
\n\
3. {}\n\
관심도 : {}\n\
감정도 : {}\n\
{}\n\
\n\
4. {}\n\
관심도 : {}\n\
감정도 : {}\n\
{}\n\
\n\
5. {}\n\
관심도 : {}\n\
감정도 : {}\n\
{}\n\
\n\
6. {}\n\
키워드 관심도 : {}\n\
키워드 감정도 : {}\n\
{}\n\
\n\
7. {}\n\
관심도 : {}\n\
감정도 : {}\n\
{}\n\
\n\
8. {}\n\
관심도 : {}\n\
감정도 : {}\n\
{}\n\
\n\
9. {}\n\
관심도 : {}\n\
감정도 : {}\n\
{}\n\
\n\
10. {}\n\
관심도 : {}\n\
감정도 : {}\n\
{}\n\
\n\
* = 당일 평균 키워드 대비 백분율\n\
\n\
📌영상으로 보고 싶으시다면?\n\
    데일리 빅데이터 유튜브!!📌\n\
    (링크)".format(topics_emoji[kr_index], topics_kr[kr_index],\
                keywords[0],keywords[1],keywords[2],keywords[3],\
                keywords[4],keywords[5],keywords[6],keywords[7],\
                keywords[8],keywords[9],\
                keywords[0],KTR[0],KTS[0],related_keywords[0],\
                keywords[1],KTR[1],KTS[1],related_keywords[1],\
                keywords[2],KTR[2],KTS[2],related_keywords[2],\
                keywords[3],KTR[3],KTS[3],related_keywords[3],\
                keywords[4],KTR[4],KTS[4],related_keywords[4],\
                keywords[5],KTR[5],KTS[5],related_keywords[5],\
                keywords[6],KTR[6],KTS[6],related_keywords[6],\
                keywords[7],KTR[7],KTS[7],related_keywords[7],\
                keywords[8],KTR[8],KTS[8],related_keywords[8],\
                keywords[9],KTR[9],KTS[9],related_keywords[9])
        
    kr_index+=1

[📰Daily Bigdata , 3/28]

[🌏오늘의 키워드🌏]

    1. 박영선
    2. 조양호
    3. 황교안
    4. 김학의
    5. 경영권
    6. 대한항공
    7. 연임
    8. 의원
    9. KT
    10. 사장


[🔎키워드 분석🔍]

1. 박영선
키워드 관심도 : 487%
키워드 감정도 : 22%
관련뉴스 : (링크) 
연관 검색 : #김의겸 #박영선 황교안 #황교안

2. 조양호
키워드 관심도 : 374%
키워드 감정도 : 46%
관련뉴스 : 링크
연관검색 : #조양호 가족 #스튜어드 십 코드 #조양호 나이

3. 황교안
키워드 관심도 : 354%
키워드 감정도 : 34%
관련뉴스 : (링크) 
연관 검색 : #박영선 황교안 #박영선 #황교안 법무부 장관

4. 김학의
키워드 관심도 : 340%
키워드 감정도 : 21%
관련뉴스 : 링크
연관검색 : #김학 의 cd #김학 의 동영상 내용 #뱀부 즐

5. 경영권
키워드 관심도 : 187%
키워드 감정도 : 63%
관련뉴스 : (링크) 
연관 검색 : #없음

6. 대한항공
키워드 관심도 : 184%
키워드 감정도 : 49%
관련뉴스 : 링크
연관검색 : #셀트리온 주가 #일 신석재 주가 #차 바이오텍

7. 연임
키워드 관심도 : 183%
키워드 감정도 : 42%
관련뉴스 : (링크) 
연관 검색 : #조양호 연임

8. 의원
키워드 관심도 : 147% 🔺
키워드 감정도 : 44% 🔻
관련뉴스 : 링크
연관검색 : #김병관 의원 #김세연 의원 #이용 주 의원

9. KT
키워드 관심도 : 143% 🔻
키워드 감정도 : 54% 🔺
관련뉴스 : (링크) 
연관 검색 : #kt 인턴 #sk 하이닉스 주식 #티 월드

10. 사장
키워드 관심도 : 133%
키워드 감정도 : 45%
관련뉴스 : 링크
연관검색 : #모래 사장 #바지 사장 #이부진 사장

[💵오늘의 증시💵]
코스피 2,230.50(+1.84)
코스닥 743.38(-3.95)
환   율 1,125.0(

In [105]:
dataframe = pd.read_csv('./Crawled Data/{}/{}_KTR_KTS_dataframe'.format(today, today))
dataframe = dataframe.sort_values(by='Total_KTR', ascending=False).iloc[0:20]
dataframe = dataframe.drop_duplicates(subset = "Keyword")
keywords = dataframe.iloc[0:10]['Keyword'].values.tolist()
dataframe

Topic Keyword  Total_KTR  Topic_KTR  KTS  \
10   politics     김의겸        451        225   27   
16   politics     김학의        430        215   13   
2     society     세월호        399        241   29   
37    foreign     대통령        327        314   19   
13   politics     나경원        249        124   29   
14   politics     대변인        234        115   20   
12   politics     황교안        216        107   35   
4     society    CCTV        200        121   64   
1     society      경찰        157         94   30   
23   economic      서울        142        206   32   
11   politics      장관        136         68   28   
35    foreign      사망        134        128   27   
9     society      불법        125         76   13   
33    foreign      중국        125        120   17   
55  entertain      검찰        119        169   49   
62     sports      선발        113        171   58   
15   politics      국회        107         53   51   
56  entertain      배우        106        150   74   
70    digital      KT        102        161   69   

                          Related_Keywords Total_KTR_change KTS_change  
10                ['#조동호', '#남경필', '#김학의']             451%        27%  
16        ['#김학의영상', '#김학의아버지', '#버닝썬vip']       430% (90🔺)   13% (8🔻)  
2         ['#박병우', '#세월호추모행사', '#세월호cctv']             399%        29%  
37   ['#리얼미터대통령지지율', '#필리핀대통령', '#고양이대통령']             327%        19%  
13         ['#나경원재산', '#나경원부동산', '#나경원리즈']             249%        29%  
14                             ['#박수현대변인']             234%        20%  
12            ['#김의겸', '#황교안고향', '#황교안나이']      216% (138🔻)   35% (1🔻)  
4                  ['#세월호cctv', '#cctv가격']             200%        64%  
1      ['#경찰교육포털', '#사복경찰', '#경찰공무원합격자발표']             157%        30%  
23  ['#서울경마출마표', '#서울우유채용', '#서울우유협동조합채용']             142%        32%  
11            ['#조명래장관', '#김의겸', '#도종환장관']       136% (59🔺)   28% (9🔻)  
35     ['#로빈윌리엄스사망', '#남보라동생사망', '#특전사사망']             134%        27%  
9       ['#정준영불법촬영', '#정준영불법동영상', '#불법웹툰']             125%        13%  
33              ['#총명탕', '#판게아', '#이순신전사']       125% (66🔺)   17% (1🔻)  
55           ['#검찰수사관', '#검찰송치란', '#검찰송치']             119%        49%  
62    ['#류현진선발경기결과', '#류현진선발경기', '#류현진경기']             113%        58%  
15      ['#국회의장임기', '#국회입법예고', '#국립중앙도서관']             107%        51%  
56         ['#모모노키카나', '#니시미야코노미', '#박초현']             106%        74%  
70     ['#kt주총', '#kt고객센터100번', '#kt골드번호']       102% (41🔻)  69% (15🔻)